osu.Game/Graphics/Cursor/OsuTooltipContainer.cs

```diff
@@ -65,10 +72,16 @@ public OsuTooltip()
                         RelativeSizeAxes = Axes.Both,
                         Alpha = 0.9f,
                     },
-                    text = new OsuSpriteText
+                    text = new TextFlowContainer(f =>
+                    {
+                        f.Font = OsuFont.GetFont(weight: FontWeight.Regular);
+                        f.Truncate = true;
+                        f.MaxWidth = max_width;
```

Comment:
This doesn't look good at all and doesn't make sense in the context of how other apps handle this.

`AllowMultiline` can't be used because it crashes on retrieving the base line height (not allowed on multiline `SpriteText` for some reason).

But, even if multiline was to be used, TFC will still not handle correctly and will put the following word always on a new line because the spritetext covers the entire horizontal area.

Basically, TFC needs a lot of work (framework-side) to do things correctly but in the meantime I'd rather not have this here because it looks wrong.

Reply:
> This doesn't look good at all and doesn't make sense in the context of how other apps handle this.
> 
> `AllowMultiline` can't be used because it crashes on retrieving the base line height (not allowed on multiline `SpriteText` for some reason).
> 
> But, even if multiline was to be used, TFC will still not handle correctly and will put the following word always on a new line because the spritetext covers the entire horizontal area.
> 
> Basically, TFC needs a lot of work (framework-side) to do things correctly but in the meantime I'd rather not have this here because it looks wrong.

I've set Truncate and MaxWidth at the same time because while the TFC handles the line-break separate by words, it doesn't consider when the word is so long like there are no spaces between the long word, so I set the max width and truncate property to the spite text so that it shows the ellipsis instead of just cut the word.

Reply:
Yes, that's not the proper way to handle it. The proper way is `AllowMultiline`, if it worked, or something equivalent implemented inside TFC, i.e., splitting by character if the line is too long.

Try it out in a browser: https://jsfiddle.net/361nLwaq/
<img width="1512" alt="image" src="https://github.com/user-attachments/assets/08ab4e70-a380-42db-b866-ba9f83ec9a99">

Reply:
> Yes, that's not the proper way to handle it. The proper way is `AllowMultiline`, if it worked, or something equivalent implemented inside TFC, i.e., splitting by character if the line is too long.
> 
> Try it out in a browser: https://jsfiddle.net/361nLwaq/ <img alt="image" width="1512" src="https://private-user-images.githubusercontent.com/1329837/354588091-08ab4e70-a380-42db-b866-ba9f83ec9a99.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI2MDA3NzIsIm5iZiI6MTcyMjYwMDQ3MiwicGF0aCI6Ii8xMzI5ODM3LzM1NDU4ODA5MS0wOGFiNGU3MC1hMzgwLTQyZGItYjg2Ni1iYTlmODNlYzlhOTkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgwMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MDJUMTIwNzUyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZmUyM2I0ZDliYjVmY2JkMjZmNzAyNGUxMGRjZjVlNmEwYWIwY2MzZjJjNzJkNDE2OTk2YjNhMWQ2ZWYyMDgxZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.yEmCmZsfXp8zEN4YqySbX4tAWFm_bgzyiAkVKZLh_N0">

So the only way to make these changes work as expected, is to fix the o!f code?

Reply:
Correct. Only wrapping per-word is fine for now because the o!f changes will not be easy to do.

Reply:
Is it okay to leave a to-do in here? For the future when the o!f fix that.

After the change: 
``` C#
Children = new Drawable[]
{
    background = new Box
    {
        RelativeSizeAxes = Axes.Both,
        Alpha = 0.9f,
    },
    text = new TextFlowContainer(f =>
    {
        f.Font = OsuFont.GetFont(weight: FontWeight.Regular);
        //TODO: when the o!f supports the AllowMultiline then this SpriteText can use that property to wrap the long one word
        //f.AllowMultiline = true;
        //f.MaxWidth = max_width;
    })
    {
        Margin = new MarginPadding(5),
        AutoSizeAxes = Axes.Both,
        MaximumSize = new Vector2(max_width, float.PositiveInfinity),
    }
};
```

Reply:
It's not going to be/can't be a `SpriteText` implementation, for one of the reasons I mentioned above. It'll be done automatically by TFC.

Reply:
The latest commit should be okay. Removed the problem code

Reply:
o!f issue: https://github.com/ppy/osu-framework/issues/5007